src/components/MoneyReportHeader.tsx

```diff
@@ -56,25 +60,35 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
     const [nextStep] = useOnyx(`${ONYXKEYS.COLLECTION.NEXT_STEP}${moneyRequestReport.reportID}`);
     const [transactionThreadReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${transactionThreadReportID}`);
     const [session] = useOnyx(ONYXKEYS.SESSION);
+    const requestParentReportAction = useMemo(() => {
+        if (!reportActions || !transactionThreadReport?.parentReportActionID) {
+            return null;
+        }
+        return reportActions.find((action) => action.reportActionID === transactionThreadReport.parentReportActionID) as OnyxTypes.ReportAction & OnyxTypes.OriginalMessageIOU;
+    }, [reportActions, transactionThreadReport?.parentReportActionID]);
+    const [transactions] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION);
```

Comment:
Connecting to all transactions seems overkill. Can we use a selector to get only the specific report transactions? I believe that we'd also be able to remove the call to `TransactionUtils.getAllReportTransactions` in `hasOnlyHeldExpenses` if we only pass report transactions to it

Reply:
Thanks for the clarification. I agree that we should address the issue with `allTransactions` in `TransactionUtils` since that won't be up to date for components using it. Given that subscribing to all transactions doesn't increase the number of re-renders, I'm fine with the current solution.